server/server_test.go

```diff
@@ -294,6 +295,33 @@ func TestMiddlewareLogging(t *testing.T) {
 	http.DefaultClient.Do(req)
 }
 
+func TestMuxMiddleware(t *testing.T) {
+	var level logging.Level
+	level.Set("info")
+	cfg := Config{
+		HTTPListenAddress: "localhost",
+		HTTPListenPort:    9193,
+		GRPCListenAddress: "localhost",
+		HTTPMiddleware:    []middleware.Interface{middleware.Logging},
+		MetricsNamespace:  "testing_mux",
+		LogLevel:          level,
+		Router:            mux.NewRouter(),
+	}
+	server, err := New(cfg)
+	require.NoError(t, err)
+
+	server.HTTP.HandleFunc("/error500", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+	})
+
+	go server.Run()
+	defer server.Shutdown()
+
+	req, err := http.NewRequest("GET", "http://127.0.0.1:9192/error500", nil)
```

Comment:
Why is this port number different to the one in the config?

Reply:
My mistake. It should be the same.